climada_petals/entity/exposures/openstreetmap/osm_dataloader.py

```diff
@@ -0,0 +1,945 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define functions to download openstreetmap data
+"""
+import geopandas as gpd
+import logging
+from osgeo import ogr, gdal
+from pathlib import Path
+import shapely
+import subprocess
+from tqdm import tqdm
+import urllib.request
+import numpy as np
+import itertools
+import time
+import overpy
+
+from climada import CONFIG
+
+LOGGER = logging.getLogger(__name__)
+gdal.SetConfigOption("OSM_CONFIG_FILE",
+                     str(Path(__file__).resolve().parent.joinpath('osmconf.ini'))) 
+
+# =============================================================================
+# Define constants
+# =============================================================================
+DATA_DIR = CONFIG.exposures.openstreetmap.local_data.dir()
+
+"""
+dictionary for the generation of the correct download api-address at 
+geofabrik.de, relating ISO3-country codes to the region & written-out name.
+Adapted from the GitHub repo osm_clipper (https://github.com/ElcoK/osm_clipper)
+Used by OSMRaw().get_data_geofabrik().
+
+Note: A few small countries will be downloaded as a multi-country file, as
+indicated in the comments.
+
+Note: "special" ISO-3 codes - Canary Islands (IC), Asian part of Russia (RUS-A),
+European part of Russia (RUS-E)
+"""
+DICT_GEOFABRIK = {
```

Comment:
Instead on initiating this dictionary in the code, you could consider to put the dictionary into the config file ([climada_petals/conf/climada.conf](https://github.com/CLIMADA-project/climada_petals/blob/feature/osm_loaders/climada_petals/conf/climada.conf)) and load it from the `CONFIG` variable, maybe as `CONFIG.exposures.openstreetmap.dict_geofabrik`.

This has the advantages that the code is shorter and easier to digest – and it allows changes in the dictionary in the local config file without changing the code of the module. However, I am not 100% sure whether it should go to config, because it is probably nothing that users would customize very frequently(?)

On the technical side, I am not sure how to write a `dict` in the config file so that it is subscriptable, so that you can call `CONFIG.exposures.openstreetmap.dict_geofabrik[iso3]`

(Here could be a solution: https://stackoverflow.com/questions/38165792/config-file-with-a-dictionary-using-python )

Reply:
Just as a remark: default values that are never to be changed can be saved as constants in 
https://github.com/CLIMADA-project/climada_python/blob/main/climada/util/constants.py

Reply:
Thanks Chahan, this could indeed be the right place for these dictionaries.

Reply:
Thanks for your suggestions! 
So, after a long back and forth, I moved both dictionaries into constants.py with commit cd2ee195.
The reasons being:
1) The geofabrik dictionaries just creates the correct urls. They will most likely hardly change. The one case where users could need to interfere is by adding countries that are on geofabrik, but not in the dict. There might be a few that are still missing (impossible to check manually).
2) The osm_cis_dict might need to be changed, if the user wants to query other key-value pairs under the critical infrastructures than those which I defined. The user can always do this with the more generic osm query functions, too, however.
3) The config file does not allow a nested dict structure (at least not easily), that can then be subscripted. (it's primarily a config object, not a dict, when loaded.). Also, I have structures of the type """Main string with 'string-inside-string' """" patterns. That's not allowed within the config style.

Reply:
this is good